nebula-logger/core/main/log-management/classes/LogEntryEventHandler.cls

```diff
@@ -66,28 +66,51 @@ public without sharing class LogEntryEventHandler extends LoggerSObjectHandler {
         List<LogEntryEvent__e> logEntryEventsToSave = new List<LogEntryEvent__e>();
         for (LogEntryEvent__e logEntryEvent : newLogEntryEvents) {
             User loggingUser = new User(Id = logEntryEvent.LoggedById__c, ProfileId = logEntryEvent.ProfileId__c);
-            LoggerSettings__c loggingUserSettings = Logger.getUserSettings(loggingUser);
-            if (loggingUserSettings.DefaultPlatformEventStorageLocation__c == DEFAULT_STORAGE_LOCATION_NAME) {
+            String platformEventStorageLocation = Logger.getUserSettings(loggingUser).DefaultPlatformEventStorageLocation__c;
+            if (logEntryEvent.TransactionScenario__c != null && LoggerScenarioRule.getAll().containsKey(logEntryEvent.TransactionScenario__c) == true) {
+                LoggerScenarioRule__mdt scenarioRule = LoggerScenarioRule.getInstance(logEntryEvent.TransactionScenario__c);
+                if (scenarioRule.IsPlatformEventStorageLocationEnabled__c == String.valueOf(true)) {
```

Comment:
Might want to store `String.valueOf(true)` outside of the for loop

Reply:
Yeah, that sounds good, that should be a quick change.